rs/bitcoin/adapter/src/get_successors_handler.rs

```diff
@@ -156,11 +165,12 @@ impl GetSuccessorsHandler {
                 .try_send(BlockchainManagerRequest::EnqueueNewBlocksToDownload(next))
                 .ok();
         }
+        obsolete_blocks.extend(request.processed_block_hashes);
```

Comment:
I'd personally find this easier to read:

```suggestion
       let mut obsolete_blocks = request.processed_block_hashes;
       if response.blocks.is_empty() {
           obsolete_block.extend(state.get_cached_blocks())
       }
```

Reply:
done